services/database/scripts.mjs

```diff
@@ -38,123 +42,173 @@ program
+  );
+  return contracts;
+}
 
-      if (options.startFrom) {
-        contractsSorted = contractsSorted.filter(
-          (obj) => obj.timestamp.getTime() > options.startFrom,
-        );
-      }
+function filterContracts(contracts, startFrom, until) {
+  let contractsSorted = contracts.sort(
+    (o1, o2) => o1.timestamp.getTime() - o2.timestamp.getTime(),
+  );
+  if (startFrom) {
+    contractsSorted = contractsSorted.filter(
+      (obj) => obj.timestamp.getTime() > startFrom,
+    );
+  }
+  if (until) {
+    contractsSorted = contractsSorted.filter(
+      (obj) => obj.timestamp.getTime() < until,
+    );
+  }
+  return contractsSorted;
+}
 
-      if (options.until) {
-        contractsSorted = contractsSorted.filter(
-          (obj) => obj.timestamp.getTime() < options.until,
-        );
-      }
+async function insertContractsToDatabase(contracts, insertFunction) {
+  console.log(
+    `Splitting ${contracts.length} contracts into chunks of ${INSERT_CHUNK_SIZE} contracts`,
+  );
+  const contractChunks = chunkArray(contracts, INSERT_CHUNK_SIZE);
+  console.log(`Number of chunks created: ${contractChunks.length}`);
+  const databasePool = new Pool({
+    host: process.env.POSTGRES_HOST,
+    port: process.env.POSTGRES_PORT,
+    database: process.env.POSTGRES_DB,
+    user: process.env.POSTGRES_USER,
+    password: process.env.POSTGRES_PASSWORD,
+    max: 5,
+  });
 
+  try {
+    for (let i = 0; i < contractChunks.length; i++) {
+      const chunk = contractChunks[i];
+      const startTime = Date.now();
+      await insertFunction(chunk, databasePool);
+      const elapsedTime = Date.now() - startTime;
       console.log(
-        `Filtered ${contracts.length} contracts to ${contractsSorted.length}`,
+        `Processed chunk ${i + 1}/${contractChunks.length} in ${elapsedTime} ms`,
       );
-    } catch (e) {
-      console.error(
-        "Error while reading and formatting files from the repository",
-      );
-      console.error(e);
-      process.exit(3);
     }
+  } catch (e) {
+    console.error("Error while storing contracts in the database", e);
+    process.exit(3);
+  } finally {
+    await databasePool.end();
+  }
+}
 
-    // 2. Insert contracts to sourcify_sync table
+async function processDirectoryPathParts(pathParts, entry) {
+  const address = pathParts.pop();
+  const chainId = pathParts.pop();
+  const matchType = pathParts.pop();
 
-    console.log("Inserting contracts to sourcify_sync table");
-    const databasePool = new Pool({
-      host: process.env.POSTGRES_HOST,
-      port: process.env.POSTGRES_PORT,
-      database: process.env.POSTGRES_DB,
-      user: process.env.POSTGRES_USER,
-      password: process.env.POSTGRES_PASSWORD,
-      max: 5,
-    });
+  if (
+    (matchType === "full_match" || matchType === "partial_match") &&
+    isNumber(chainId)
+  ) {
+    return { chainId, address, timestamp: entry.stats.birthtime, matchType };
+  }
+  return null;
```

Comment:
I think it shouldn't go unnoticed if there's an unexpected type (i.e. folder name) inside the `contracts/`. This is an assumption we are making and if it doesn't hold the program should not work. If we just log it will never be noticed.